config/build_simulation.py

```diff
@@ -0,0 +1,183 @@
+    with open(network_path) as user_file:
+        file_contents = user_file.read()
+    network_json = json.loads(file_contents)
+    network_name = list(network_json.keys())[0]
+    network_json = list(network_json.values())[0]
+    network = json_graph.node_link_graph(network_json,
+                                         directed=network_json.get("directed", False),
+                                         multigraph=network_json.get("multigraph", False))
+    return PowerNetwork(network_name, network, weight_factor)
+
+
+# Actor
+def create_actor_from_config(actor_id, asset_dict={}, start_date="2016-01-01", nb_ts=None,
+                             ts_hour=1, cols=["load", "pv", "schedule", "prices"]):
+    df = pd.DataFrame([], columns=cols)
+    start_date, end_date = end_date_from_ts(start_date, nb_ts, ts_hour)
+
+    # Read csv files for each asset
+    csv_peak = {}
+    for col, csv_dict in asset_dict.items():
+        # if csv_dict is empty
+        if not csv_dict:
+            continue
+        csv_df = pd.read_csv(csv_dict["csv"], sep=',', parse_dates=['Time'], dayfirst=True,
+                             index_col=['Time'])
+        df.loc[:, col] = csv_df[start_date:end_date]
+        # Save peak value and normalize time series
+        csv_peak[col] = df[col].max()
+
+    df = basic_strategy(df)
+
+    return Actor(actor_id, df)
+
+
+def read_config_json(config_json):
+    config_df = pd.read_json(config_json)
+    # Include market maker
+    if 'market_maker' in list(config_df['prosumerType']):
+        config_df = config_df[config_df.prosumerType != 'market_maker']
+        market_maker_sell = pd.DataFrame({'prosumerName': 'market_maker_sell',
+                                          'prosumerType': 'market_maker_sell', 'gridLocation':
+                                              'market_maker'}, index=[0])
+        market_maker_buy = pd.DataFrame({'prosumerName': 'market_maker_buy',
+                                         'prosumerType': 'market_maker_buy',
+                                         'gridLocation': 'market_maker'}, index=[0])
+        config_df = pd.concat([config_df, market_maker_buy, market_maker_sell], ignore_index=True)
+
+    return config_df
+
+
+# Scenario
+def create_scenario_from_config(config_json, network_path, loads_dir_path, data_dirpath=None,
+                                weight_factor=1, ts_hour=4, nb_ts=None, start_date="2016-01-01",
+                                plot_network=False, price_filename="basic_prices.csv"):
+    # Extend paths
+    loads_path = data_dirpath.joinpath("load")
+    pv_path = data_dirpath.joinpath("pv")
+    price_path = data_dirpath.joinpath("price")
+
+    # Parse json
+    config_df = read_config_json(config_json)
+    # Create nodes for power network
+    pn = create_power_network_from_config(network_path, weight_factor)
+
+    if plot_network is True:
+        pn.plot()
+
+    actors = []
+    file_dict = {}
+    asset_dict = {}
```

Comment:
dictionaries have to be reseted for each actor

Reply:
... accordingly the asset data has to be set to zero or the actor has to interpret the nan values as such

Reply:
Don't know how I missed this! Thanks